### PR TITLE
Update to 5.4

### PR DESCRIPTION
**Pull request by mtdavidson**
Includes required change for logstash config and disables the new machine learning functionality in x-pack by default.

### CODE REVIEW

File: logstash/config/logstash.yml [hunk 1]
```diff
@@ -9,3 +9,8 @@ http.host: "0.0.0.0"
 ##     https://www.elastic.co/guide/en/x-pack/current/installing-xpack.html#xpack-enabling
 #
 xpack.monitoring.enabled: false
+
+## The path to the Logstash config for the main pipeline.
+## see https://www.elastic.co/guide/en/logstash/current/logstash-settings-file.html
+#
+path.config: /usr/share/logstash/pipeline
```
Comment (antoineco) [inline logstash/config/logstash.yml]: Already defaults to `/usr/share/logstash/pipeline` so this should be removed. See https://www.elastic.co/guide/en/logstash/current/dir-layout.html

File: docker-compose.yml [hunk 2]
```diff
@@ -14,6 +14,7 @@ services:
       #     https://www.elastic.co/guide/en/x-pack/current/installing-xpack.html#xpack-enabling
       xpack.security.enabled: "false"
       xpack.monitoring.enabled: "false"
+      xpack.ml.enabled: "false"
```
Comment (antoineco) [inline docker-compose.yml]: Should also be disabled in the [`kibana.yml`](https://github.com/deviantony/docker-elk/blob/52f43a4a72dbfef0a0fe604bad384027be7c9faa/kibana/config/kibana.yml) file for completeness. See https://www.elastic.co/guide/en/x-pack/current/installing-xpack.html#xpack-enabling

File: logstash/config/logstash.yml (hunk 1 above)
Comment (mtdavidson) [inline logstash/config/logstash.yml]: That was my thought as well but running it you get 

```
logstash_1       | Sending Logstash's logs to /usr/share/logstash/logs which is now configured via log4j2.properties
logstash_1       | [2017-05-08T15:02:53,112][INFO ][logstash.setting.writabledirectory] Creating directory {:setting=>"path.queue", :path=>"/usr/share/logstash/data/queue"}
logstash_1       | usage:
logstash_1       |   bin/logstash -f CONFIG_PATH [-t] [-r] [] [-w COUNT] [-l LOG]
logstash_1       |   bin/logstash -e CONFIG_STR [-t] [--log.level fatal|error|warn|info|debug|trace] [-w COUNT] [-l LOG]
logstash_1       |   bin/logstash -i SHELL [--log.level fatal|error|warn|info|debug|trace]
logstash_1       |   bin/logstash -V [--log.level fatal|error|warn|info|debug|trace]
logstash_1       |   bin/logstash --help
logstash_1       | ERROR: No configuration file was specified. Perhaps you forgot to provide the '-f yourlogstash.conf' flag?
dockerelk_logstash_1 exited with code 1
```

File: docker-compose.yml (hunk 2 above)
Comment (mtdavidson) [inline docker-compose.yml]: Ah yes I missed that will get it corrected

File: logstash/config/logstash.yml (hunk 1 above)
Comment (antoineco) [inline logstash/config/logstash.yml]: Oh sorry! It was added as part of elastic/logstash-docker@303eafecbd84753d42000772aafa14c1d1eec1ff, you're right. Can you just delete the comment then?

**Comment by antoineco** [comment]
Thanks a lot @mtdavidson !

**Comment by mtdavidson** [comment]
No problem thanks for the quick merge :)

**Comment by antoineco** [comment]
The release notes, just for reference:

- https://www.elastic.co/guide/en/elasticsearch/reference/5.4/release-notes-5.4.0.html
- https://www.elastic.co/guide/en/kibana/5.4/release-notes-5.4.0.html
- https://www.elastic.co/guide/en/logstash/5.4/logstash-5-4-0.html

**Pull request merged**